src/structures/Widget.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const Base = require('./Base');
+const Collection = require('../util/Collection');
+
+/**
+ * Represents a Widget.
+ */
+class Widget extends Base {
+  /**
+   * @param {Client} client - The instantiating client
+   * @param {Object} data - The raw data;
+   */
+  constructor(client, data) {
+    super(client);
+    this._patch(data);
+  }
+  /**
+   * Builds the widget with the provided data.
+   * @param {*} data The raw data of the widget
+   * @private
+   */
+  _patch(data) {
+    /**
+     * When a widget is disable, data will always send a code. When it's enable, data doesn't have a code.
+     */
+    if ('code' in data) {
+      /**
+       * The code provide with the data.
+       * @type {?number}
+       */
+      this.code = data.code;
+
+      /**
+       * The error message.
+       * @type {?string}
+       */
+      this.message = data.message;
+
+      /**
+       * Check if the guild has the widget enabled or not.
+       * @type {boolean}
+       */
+      this.enabled = false;
+    } else {
+      /**
+       * The id of the guild.
+       * @type {?string}
+       */
+      this.id = data.id;
+
+      /**
+       * The name of the guild.
+       * @type {?string}
+       */
+      this.name = data.name;
+
+      /**
+       * The invite of the guild.
+       * @type {?string | ?null}
```

Comment:
Why would it be an optional null if not a string?

Reply:
That's weird I know but I need to explain
When the raw object comes, invite can appears or not due to the error message and the code
![image](https://user-images.githubusercontent.com/42680097/118074796-fde11480-b3ae-11eb-94d8-6c6018faa83f.png)
But "instant_invite" can be null too when the widget is enabled :
![image](https://user-images.githubusercontent.com/42680097/118074727-d722de00-b3ae-11eb-85e0-06129faddf44.png)

Reply:
Yeah but as I said in my other comment, that first screenshot is a JSON error, not a widget. You shouldn't create one at all if you get an error code.